python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py

```diff
@@ -0,0 +1,491 @@
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+    evaluations = []
+    sagemaker_client = get_client('sagemaker', event)
+    endpoint_config_list = get_all_endpoint_config_names(sagemaker_client)
+    #SCE 2
+    if not endpoint_config_list:
+        return "NOT_APPLICABLE"
+
+    for endpoint_config in endpoint_config_list:
+        sagemaker_endpoint_details = sagemaker_client.describe_endpoint_config(EndpointConfigName=endpoint_config)
+
+        if 'KmsKeyId' in sagemaker_endpoint_details:
+
+            #SCENARIO 6: KmsKey sepcified in sagemaker endpoint and no rule parameters
+            if not valid_rule_parameters:
+                evaluations.append(build_evaluation(endpoint_config, 'COMPLIANT', event))
+
+            #SCENARIO 5: KmsKey specified in the sagemaker endpoint and it matches with the rule parameters
+            elif sagemaker_endpoint_details['KmsKeyId'] in valid_rule_parameters:
+                evaluations.append(build_evaluation(endpoint_config, 'COMPLIANT', event))
+
+            #SCENARIO 4: KmsKey specified in the sagemaker endpoint and does not it matches with the rule parameters
+            else:
+                evaluations.append(build_evaluation(endpoint_config, 'NON_COMPLIANT', event, annotation="AWS KMS Key configured for this Amazon SageMaker Endpoint Config is not an KMS Key allowed in the rule parameter (keyIds)"))
+
+        #SCENARIO 3: KmsKey is not specified in the sagemaker endpoint
+        else:
+            evaluations.append(build_evaluation(endpoint_config, 'NON_COMPLIANT', event, annotation="No AWS KMS Key is configured for this Amazon SageMaker Endpoint Config."))
+
+    return evaluations
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    if not rule_parameters:
+        return {}
+
+    if not 'keyIds' in rule_parameters:
+        return {}
+
+    given_keyids = rule_parameters['keyIds'].replace(' ', '').split(',')
+    valid_keyids = []
+

```

Comment:
initialise this at the start of the function and return it as it is, i.e. blank:
```
valid_key_ids = []
if "keyArns" in rule_parameters:
    ...
    valid_key_ids.append()
    ...
return valid_key_ids
```

Reply:
@Geeth26 The evaluate_parameters method needs to be modified as per above comment